### PR TITLE
Add a brief "basic usage" document

### PR DESCRIPTION
**Pull request by mkouba**
This is the first attempt to document the basic usage of Gizmo API and resolve https://github.com/quarkusio/gizmo/issues/124.

### CODE REVIEW

File: USAGE.adoc [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+<1> Create a method with the following signature `private boolean alwaysReturnFalse()`.
+<2> A `MethodCreator` must always define a return statement.
+<3> This is an equivalent of `return false;`.
+
+[[method_bytecode]]
+== Generating Method Bytecode
+
+The `BytecodeCreator` interface allows you to generate some most common bytecode operations.
+It's not intended to cover all the functionality provided by JVM and Java language though.
+Most operations return and accept a `ResultHandle`.
```
Comment (mkouba) [inline USAGE.adoc:76]: I'm not so sure about this description of `ResultHandle` but I didn't find a better words... so feel free to reword my initial attempt.

File: USAGE.adoc [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+  myField.setModifiers(ACC_FINAL | ACC_PROTECTED); <2>
+}
+----
+<1> Creates a field `score` of type `Integer`.
+<2> Makes the field `protected final`.
+
+== Adding Methods
+
+A method can be added through a `MethodCreator` interface returned from the `ClassCreator#getMethodCreator()` methods.
+By default, the generated field is `public`.
```
Comment (Ladicek) [inline USAGE.adoc:49]:
```suggestion
By default, the generated method is `public`.
```

File: USAGE.adoc [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+You can use the `MethodCreator#setModifiers()` method to control the modifiers of the generated field.
```
Comment (Ladicek) [inline USAGE.adoc:50]:
```suggestion
You can use the `MethodCreator#setModifiers()` method to control the modifiers of the generated method.
```

File: USAGE.adoc (hunk 1 above)
Comment (Ladicek) [reply USAGE.adoc:76]: I think your explanation makes sense, even if it is factually incorrect.

(Gizmo allocates a local variable for almost all `ResultHandle`s, with the exception of a top-of-stack value that is a direct input to the immediately following operation, which is kept on stack and not spilled into a local.)

File: USAGE.adoc [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+void iterate(MethodCreator method, ResultHandle iterable) {
+    // for (Object element : iterable) { 
+    //   System.out.println(element);
+    // }
+    ForEachLoop loop = method.forEach(list);
+    BytecodeCreator block = loop.block();
+    Gizmo.systemOutPrintln(block, loop.element());
+}
+----
+
```
Comment (Ladicek) [inline USAGE.adoc:185]: Maybe add a note here that `ForEachLoop` only works for `Iterable`s and not for arrays?

**Review by Ladicek — APPROVED** [review]
Very nice!

**Comment by Sanne** [comment]
Love it, thank you! I only need gizmo rarely, and need to find examples each time as w/o practice I forget.. so this is very useful to me.

**Pull request merged**